utils/installer/install.sh

```diff
@@ -1,225 +1,191 @@
-
-# install pynvim
-(pip3 list | grep pynvim >/dev/null && echo "pynvim installed, moving on...") || installpynvim
-
-if [ -e "$HOME/.local/share/lunarvim/site/pack/packer/start/packer.nvim" ]; then
-	echo 'packer already installed'
-else
-	installpacker
-fi
-
-if [ -e "$HOME/.local/share/lunarvim/lvim/init.lua" ]; then
-	echo 'LunarVim already installed'
-else
-	# clone config down
-	cloneconfig "$@"
-	# echo 'export PATH=$HOME/.config/nvim/utils/bin:$PATH' >>~/.zshrc
-	# echo 'export PATH=$HOME/.config/lunarvim/utils/bin:$PATH' >>~/.bashrc
-fi
-
-echo "I recommend you also install and activate a font from here: https://github.com/ryanoasis/nerd-fonts"
-# echo 'export PATH=/home/$USER/.config/lunarvim/utils/bin:$PATH appending to zshrc/bashrc'
+declare -r LVBRANCH="${LVBRANCH:-master}"
+declare -r LV_REPO="${LV_REPO:-\"ChristianChiarulli/lunarvim.git\"}"
+
+declare -r XDG_DATA_HOME=${XDG_DATA_HOME:-"$HOME/.local/share"}
+declare -r XDG_CACHE_HOME=${XDG_CACHE_HOME:-"$HOME/.cache"}
+declare -r XDG_CONFIG_HOME=${XDG_CONFIG_HOME:-"$HOME/.config"}
+
+# declare -r NEOVIM_RUNTIME_DIR="$XDG_DATA_HOME/nvim"
+# declare -r NEOVIM_CONFIG_DIR="$XDG_CONFIG_HOME/nvim"
+# declare -r NEOVIM_CACHE_DIR="$XDG_CACHE_HOME/nvim"
+
+declare -r LUNARVIM_RUNTIME_DIR="$XDG_DATA_HOME/lunarvim"
+declare -r LUNARVIM_CONFIG_DIR="$XDG_CONFIG_HOME/lvim"
+
+# declare -a __nvim_dirs=(
+#   "$NEOVIM_CACHE_DIR"
+#   "$NEOVIM_CONFIG_DIR"
+#   "$NEOVIM_RUNTIME_DIR"
+# )
+
+declare -a __lvim_dirs=(
+  "$LUNARVIM_CONFIG_DIR"
+  "$LUNARVIM_RUNTIME_DIR"
+)
+
+declare -a __npm_deps=(
+  "neovim"
+  "tree-sitter-cli"
+)
+
+declare -a __pip_deps=(
+  "pynvim"
+  "neovim-remote"
+)
+
+function main() {
+  # Welcome
+  echo "Installing LunarVim"
+
+  echo "Detecting platform for managing any additional neovim dependencies"
+  detect_platform
+  echo "Would you like to check noevim's nodejs dependencies?"
+  read -p "[y]es or [n]no (default: no) : " -r answer
+  [ "$answer" != "${answer#[Yy]}" ] && install_npm_deps
+  echo "Would you like to check noevim's python dependencies?"
+  read -p "[y]es or [n]no (default: no) : " -r answer
+  [ "$answer" != "${answer#[Yy]}" ] && install_pip_deps
+  case "$@" in
+    *--overwrite*)
+      echo "!!Warning!! -> Removing all lunarvim related config \
+        because of the --overwrite flag"
+      read -p "Would you like to continue? [y]es or [n]o : " -r answer
+      [ "$answer" == "${answer#[Yy]}" ] && exit 1
+      for dir in "${__lvim_dirs[@]}"; do
+        [ -d "$dir" ] && rm -rf "$dir"
+      done
+      ;;
+  esac
+
+  if [ -e "$LUNARVIM_RUNTIME_DIR/site/pack/packer/start/packer.nvim" ]; then
+    echo "packer already installed"
+  else
+    install_packer
+  fi
+
+  if [ -e "$LUNARVIM_RUNTIME_DIR/lvim/init.lua" ]; then
+    echo "LunarVim already installed"
+    echo "checking for the latest updates instead"
+    local CMD="git -C $LUNARVIM_RUNTIME_DIR/lvim pull origin $LVBRANCH"
```

Comment:
Should this case perhaps complain if the existing remote doesn't match the expected?

Reply:
You mean if it's not called "origin" right? damn that got quite a bit more complicated just now. I should create a function for it which will actually prompt the user before doing anything to the repo.

Reply:
Yeah, or if the supplied LV_REPO isn't the same as origin

Reply:
Yeah this is way too complicated. I'm gonna leave this for a separate PR with regards on how to handle updating lunarvim. 
There are simply way too many possible cases so it's not trivial. I guess we could open a poll and see what people think the workflow should look like.